docs/book/src/faq.md

```diff
@@ -0,0 +1,88 @@
+
+# Kubebuilder FAQ
+
+## How to undestand the domain field?
+First of all, we should get familiar with the GVK terminology, check [here](./cronjob-tutorial/gvks.md).
+
+The domain is for the group suffix, to explicitly show the resource group category.
+For example, if set `--domain=example.com`:
+```
+kubebuilder init --domain example.com --repo xxx --plugins=go/v4-alpha
+kubebuilder create api --group mygroup --version v1beta1 --kind Mykind
+```
+Then the result resource group will be `mygroup.example.com`.
+
+If domain field not set, the default value is `my.domain`.
+
+## How to validate API or set default value?
+We can use `webhook` to achieve it. See [webhook](./reference/webhook-overview.md)
+
+For example, we can create webhook with defaulting and validating logic by:
+```
+kubebuilder create webhook --group mygroup --version v1beta1 --kind Mykind --defaulting --programmatic-validation
+```
+Then the generated webhook code is like this:
+```go
+func (r *Mykind) Default() {
+	// TODO(user): fill in your defaulting logic.
+}
+
+func (r *Mykind) ValidateCreate() error {
+	// TODO(user): fill in your validation logic upon object creation.
+	return nil
+}
+
+func (r *Mykind) ValidateUpdate(old runtime.Object) error {
+	// TODO(user): fill in your validation logic upon object update.
+	return nil
+}
+
+func (r *Mykind) ValidateDelete() error {
+	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+```
+So it is easy to add API default and validate logic.
+
+## How to use `klog` or other loggers as the project logger?
+We can set the project logger by:
```

Comment:
See that we follow the same docs guidelines applied for k8s docs. Therefore, the proteome here seems that should be you and not we. Also, we need to describe where the change needs to be done. So, wdyt:

```suggestion
Into the `main.go` you can replace:

<add code>

with:

<add code example>
```

Reply:
Also, I think we need here explain the options.